test/framework/control_plane.go

```diff
@@ -23,42 +23,51 @@ import (
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/api/core/v1"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/util/wait"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
 	cabpkv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1alpha3"
 	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	// eventuallyInterval is the polling interval used by gomega.Eventually
+	eventuallyInterval = 10 * time.Second
+)
+
 // ControlplaneClusterInput defines the necessary dependencies to run a multi-node control plane cluster.
 type ControlplaneClusterInput struct {
-	Management    ManagementCluster
-	Cluster       *clusterv1.Cluster
-	InfraCluster  runtime.Object
-	Nodes         []Node
-	CreateTimeout time.Duration
-	DeleteTimeout time.Duration
+	Management                  ManagementCluster
+	Cluster                     *clusterv1.Cluster
+	InfraCluster                runtime.Object
+	PrimaryControlPlaneNode     Node
+	AdditionalControlPlaneNodes []Node
+	MachineDeployment           MachineDeployment
+	RelatedResources            []runtime.Object
```

Comment:
What's the use case for the related resources? Not opposed, just curious what you're using it for.

Reply:
`HAProxyLoadBalancer` in my case

Reply:
ahhh 👍 that makes sense, thanks